### DISCUSSION TITLE
Initializing the backend in production

### DISCUSSION THREAD

**Discussion by machov** [fastapi/full-stack-fastapi-template#1177] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched in the documentation/README.
- [X] I already searched in Google "How to do X" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs/README and didn't find an answer.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```Text
`uvicorn main:app --reload`
```

### Description

I'm struggling to understand the sequence of commands in backend dockerfile

the Dockerfile runs `uvicorn main:app --reload` when importing the Dockerfile from [FROM tiangolo/uvicorn-gunicorn-fastapi:python3.10
](https://github.com/tiangolo/uvicorn-gunicorn-docker/blob/7ff55d0d05ab9516f69fbefa69dc651612fd45fe/docker-images/python3.9-slim.dockerfile#L28C1-L28C18) with command `CMD ["/start.sh"]` is there!

however `[COPY ./app /app/app](https://github.com/tiangolo/full-stack-fastapi-template/blob/master/backend/Dockerfile#L28) takes place till the end here .

How is  unicorn being executed without the app? does this mean CMD at the end of `tiangolo/uvicorn-gunicorn-fastapi:python3.10` is somehow delayed?

I see that `docker-compose.override.yml` has 
    command: [/start-reload.sh](https://github.com/tiangolo/uvicorn-gunicorn-docker/blob/7ff55d0d05ab9516f69fbefa69dc651612fd45fe/docker-images/start-reload.sh)
which executes uvicorn

however I can't understand how does `docker-compose.yml` work in production

### Operating System

macOS

### Operating System Details

docker

### Python Version

3.9

**Comment by YuriiMotov** [comment, marked as answer]
In current implementation, there is a command defined in in `.Dockerfile`:
```
CMD ["fastapi", "run", "--workers", "4", "app/main.py"]`
```

And for development, this command is overridden in `docker-compose.override.yml`:
```
  backend:
    ...
    command:
      - fastapi
      - run
      - --reload
      - "app/main.py"
```

So, in production, the command defined in `.Dockerfile` is used
